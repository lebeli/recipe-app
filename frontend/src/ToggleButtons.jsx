import React from "react";
import Box from "@material-ui/core/Box";
import FilterCategory from "./FilterCategory";

import "./Filter.scss";

function ToggleButtons(params) {
  return (
    <div className="ToggleButtons">
      <Box display="flex" justifyContent="center" m={1} p={1}>
        <Box p={1}>
          <FilterCategory
            categoryName="Frühstück"
            value="breakfast"
            selected={params.breakfast}
            updateState={params.updateState}
          />
        </Box>
        <Box p={1}>
          <FilterCategory
            categoryName="Mittagessen"
            value="lunch"
            selected={params.lunch}
            updateState={params.updateState}
          />
        </Box>
        <Box p={1}>
          <FilterCategory
            categoryName="Abendessen"
            value="dinner"
            selected={params.dinner}
            updateState={params.updateState}
          />
        </Box>
      </Box>
      <Box display="flex" justifyContent="center" m={1} p={1}>
        <Box p={1}>
          <FilterCategory
            categoryName="vegetarisch"
            value="vegetarian"
            selected={params.vegetarian}
            updateState={params.updateState}
          />
        </Box>
        <Box p={1}>
          <FilterCategory
            categoryName="vegan"
            value="vegan"
            selected={params.vegan}
            updateState={params.updateState}
          />
        </Box>
        <Box p={1}>
          <FilterCategory
            categoryName="schnell"
            value="schnell"
            selected={params.schnell}
            updateState={params.updateState}
          />
        </Box>
        <Box p={1}>
          <FilterCategory
            categoryName="ich habe Zeit"
            value="i_have_time"
            selected={params.i_have_time}
            updateState={params.updateState}
          />
        </Box>
      </Box>
    </div>
  );
}

export default ToggleButtons;
