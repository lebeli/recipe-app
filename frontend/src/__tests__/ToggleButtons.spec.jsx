import React from "react";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import ToggleButtons from "../ToggleButtons";

describe("<ToggleButtons />", () => {
  test("should render elements", () => {
    const wrapper = mount(
      <ToggleButtons
        breakfast={false}
        lunch={false}
        dinner={false}
        vegetarian={false}
        vegan={false}
        fast={false}
        i_have_time={false}
        chosen_ingredients={[]}
      />
    );

    expect(wrapper.find(".ToggleButtons").length).toEqual(1);
    expect(wrapper.find(".FilterCategory").length).toEqual(7);
  });

  test("renders correctly", () => {
    const tree = renderer
      .create(
        <ToggleButtons
          breakfast={false}
          lunch={false}
          dinner={false}
          vegetarian={false}
          vegan={false}
          fast={false}
          i_have_time={false}
          chosen_ingredients={[]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
