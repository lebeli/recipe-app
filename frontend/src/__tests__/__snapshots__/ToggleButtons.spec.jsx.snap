// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleButtons /> renders correctly 1`] = `
<div
  className="ToggleButtons"
>
  <div
    className="MuiBox-root MuiBox-root-30"
  >
    <div
      className="MuiBox-root MuiBox-root-31"
    >
      <div
        className="FilterCategory"
      >
        <button
          aria-pressed={false}
          className="MuiButtonBase-root MuiToggleButton-root"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
          value="breakfast"
        >
          <span
            className="MuiToggleButton-label"
          >
            Frühstück
          </span>
        </button>
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-32"
    >
      <div
        className="FilterCategory"
      >
        <button
          aria-pressed={false}
          className="MuiButtonBase-root MuiToggleButton-root"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
          value="lunch"
        >
          <span
            className="MuiToggleButton-label"
          >
            Mittagessen
          </span>
        </button>
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-33"
    >
      <div
        className="FilterCategory"
      >
        <button
          aria-pressed={false}
          className="MuiButtonBase-root MuiToggleButton-root"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
          value="dinner"
        >
          <span
            className="MuiToggleButton-label"
          >
            Abendessen
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root MuiBox-root-34"
  >
    <div
      className="MuiBox-root MuiBox-root-35"
    >
      <div
        className="FilterCategory"
      >
        <button
          aria-pressed={false}
          className="MuiButtonBase-root MuiToggleButton-root"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
          value="vegetarian"
        >
          <span
            className="MuiToggleButton-label"
          >
            vegetarisch
          </span>
        </button>
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-36"
    >
      <div
        className="FilterCategory"
      >
        <button
          aria-pressed={false}
          className="MuiButtonBase-root MuiToggleButton-root"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
          value="vegan"
        >
          <span
            className="MuiToggleButton-label"
          >
            vegan
          </span>
        </button>
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-37"
    >
      <div
        className="FilterCategory"
      >
        <button
          aria-pressed={false}
          className="MuiButtonBase-root MuiToggleButton-root"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
          value="fast"
        >
          <span
            className="MuiToggleButton-label"
          >
            schnell
          </span>
        </button>
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-38"
    >
      <div
        className="FilterCategory"
      >
        <button
          aria-pressed={false}
          className="MuiButtonBase-root MuiToggleButton-root"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
          value="i_have_time"
        >
          <span
            className="MuiToggleButton-label"
          >
            ich habe Zeit
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
